### PR TITLE
feat(appsync): support AppSync Events

### LINKED ISSUE #32004: appsync: L2 for AppSync Events
### Describe the feature

AWS AppSync Events allows developers to easily broadcast real-time event data to subscribers using serverless WebSocket APIs, eliminating the need to build WebSocket infrastructure. 

Release blog:
https://aws.amazon.com/blogs/mobile/announcing-aws-appsync-events-serverless-websocket-apis/

Developer Guide:
https://docs.aws.amazon.com/appsync/latest/eventapi/event-api-welcome.html

### Use Case

Easily configure AppSync Events resources, e.g.

* Create an Event API
* Update configuration of the API, such as
  * [authorization](https://docs.aws.amazon.com/appsync/latest/eventapi/configure-event-api-auth.html)
  * [namespace](https://docs.aws.amazon.com/appsync/latest/eventapi/channel-namespaces.html)
  * [logging](https://docs.aws.amazon.com/appsync/latest/eventapi/event-api-monitoring.html)
  * [custom domain](https://docs.aws.amazon.com/appsync/latest/eventapi/event-api-custom-domains.html)
* Configure [event handlers](https://docs.aws.amazon.com/appsync/latest/eventapi/channel-namespace-handlers.html) for a namespace

### Proposed Solution

_No response_

### Other Information

~~It seems we don't have [CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/AWS_AppSync.html) for AppSync Events support yet.~~ After L1s are added, we can design L2 APIs. edit) L1s appears merged recently: https://github.com/aws/aws-cdk/pull/32007
Here is an implementation example: https://github.com/aws-samples/websocket-api-cognito-auth-sample/blob/main/cdk/lib/construct/events.ts

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.165.0

### Environment details (OS name and version, etc.)

macOS

### PR DESCRIPTION
**Pull request by mazyu36**
### Issue # (if applicable)

Closes #32004.

### Reason for this change
To support AppSync Event APIs.

### Description of changes
Add `Api` and `ChannelNamespace` Construct.

### Description of how you validated changes
Add unit tests and an integ test.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-appsync/lib/auth-config.ts [hunk 1]
```diff
@@ -0,0 +1,230 @@
+import { Construct } from 'constructs';
+import { CfnApiKey } from './appsync.generated';
+import { IUserPool } from '../../aws-cognito';
+import { IFunction } from '../../aws-lambda';
+import { Duration, Expiration } from '../../core';
+
+/**
+ * enum with all possible values for AppSync authorization type
+ */
+export enum AuthorizationType {
```
Comment (mazyu36) [inline packages/aws-cdk-lib/aws-appsync/lib/auth-config.ts:10]: The authorization config is largely shared between GraphQL and Events, so it has been extracted into a separate file.

File: packages/aws-cdk-lib/aws-appsync/lib/channel-namespace.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    return new Import(scope, id);
+  }
+
+  /**
+   * The ARN of the AppSync channel namespace
+   */
+  public readonly channelNamespaceArn: string;
+
+  constructor(scope: Construct, id: string, props: ChannelNamespaceProps) {
+    if (props.channelNamespaceName !== undefined && !Token.isUnresolved(props.channelNamespaceName)) {
```
Comment (mazyu36) [inline packages/aws-cdk-lib/aws-appsync/lib/channel-namespace.ts:88]: `ChannelNamespaceName`'s pattern is quoted from Management Console.
image

Currently, this pattern is inconsistent with the one described in the CloudFormation documentation.

I've reported it. https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/2208

**Comment by onlybakam** [comment]
hey @mazyu36, thanks for opening this PR. the AppSync team is working on an L2 implementation as well. It would be good to collaborate and merge some of those efforts. Let me know if you're open to that and we can connect.

**Comment by mazyu36** [comment]
@onlybakam 

Thank you.
Sounds great! I'd love to collaborate.

**Comment by onlybakam** [comment]
cool. can you email me at `bricep`@`amazon.com` ?

**Comment by mazyu36** [comment]
Since I am currently collaborating with the AppSync team, this PR will remain in Draft status for the time being.

**Comment by tai-acall** [comment]
> Since I am currently collaborating with the AppSync team, this PR will remain in Draft status for the time being.

wow ! I can't wait to see this, is any schedule about release?

I'm considering how to do implementation for AppSnyc Events in our code base, if L2 construct ready will be so much helpful  !

@mazyu36

**Comment by mazyu36** [comment]
@tai-acall 
Thank you for your comment.
I cannot provide specific details about the schedule, but I believe an update will come soon.

Please be patient for a moment.​​​​​​​​​​​​​​​​

**Comment by mazyu36** [comment]
New PR by AppSync Team is now opened.

#32505

So I close this PR.

**Pull request closed without merging**